### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by fatima-najafi**
### OOP school library: add Person, Student and Teacher classes
 In this PR , I implement the following requirement:
- [x] Create class Person with the following:
- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- [x] Create class Student with the following:
- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

- [x] Create class Teacher with the following:

- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.
- [x] Each class  saved in a separate file.

### CODE REVIEW

**Review by julie-ify — COMMENTED** [review]
Hi @fatima-najafi, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Kindly fix your readme file. It is for the enumerable project but this is a different project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Permission is hereby granted, free of charge, to any person obtaining a copy of this [Application] and associated documentation files, to deal in the [Application] without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the [Application], and to permit persons to whom the [Application] is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the [Application].
+
+THE [Application] IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE [Application] OR THE USE OR OTHER DEALINGS IN THE [Application].
+## Copyright 2023, Fatima Najfi, Ochieng Steven and Erick Bupilipili
```
Comment (zeff96) [inline LICENSE]: - [ ] Hello! 😊 While it's awesome to utilize custom templates, it's crucial to tailor them to precisely fit the requirements of your specific project. 👌 Take a moment to examine your LICENSE file closely, and let's give it a personalized touch in accordance with the project's needs. 📄💼 Pay meticulous attention to detail to steer clear of minor errors that can give the file an unprofessional appearance. 🧐🚀

File: src/person.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Person
+  def initialize(name = 'Unknown', age = 0, parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id
```
Comment (zeff96) [inline src/person.rb]: - [  OPTIONAL ] Hey there! 🚀 I was thinking, wouldn't it be awesome if we moved those getters and setters right up to the top of the class? 📦🚀 What do you reckon? 🤔 Let's aim to declare the getters and setters at the top of the class for maximum clarity and organization! 🌟👩‍💻🌈

File: src/student.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'Person'
+
+class Student < Person
+  def initialize(name, age, classroom)
+    super(name, age)
```
Comment (zeff96) [inline src/student.rb]: - [ ]  Hello! 📝✨ It's important to note that the Student class inherits from the Person class, which means it should inherit all the parameters, including parent_permission with a default value of true. Additionally, the name parameter should have a default value of 'unknown'. 🚀📚 This ensures that our Student class inherits the necessary attributes from its parent class while also providing meaningful defaults where needed. 🧬👨‍👧‍👦📖

File: src/teacher.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require_relative 'Person'
+
+class Teacher < Person
+  def initialize(name, age, specialization)
```
Comment (zeff96) [inline src/teacher.rb]: - [ ]  Hello! 📝✨ It's important to note that the Student class inherits from the Person class, which means it should inherit all the parameters, including parent_permission with a default value of true. Additionally, the name parameter should have a default value of 'unknown'. 🚀📚 This ensures that our Student class inherits the necessary attributes from its parent class while also providing meaningful defaults where needed. 🧬👨‍👧‍👦📖

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @fatima-najafi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
✔️ Correct GitFlow
✔️ Descriptive pull request
✔️ Linters passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED
Hi @fatima-najafi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats

## Highlights
✔️ Descriptive pull request
✔️ Correct GitFlow
✔️ Neat and clean code, keep the good work
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**